File: roundtrip/relay/lib/edit-policy.ts
import policy from './edit-policy.json';
export const examples = policy.examples;
export const scopeMessage = policy.message;
// Fail closed: every clause must describe an allowed adjustment or preservation.
// This intentionally narrow grammar is a demo boundary, not an AI moderation claim.
export function supportedEdit(feedback: string): boolean {
  const clauses=feedback.toLowerCase().trim().replace(/^(?:please|just for proof of concept)\s+/,'').split(/[.!;]+|,?\s+but\s+|,\s*(?=keep\b)|\s+and\s+(?=crop\b)/).map(s=>s.trim()).filter(Boolean);
  if(!clauses.length||clauses.length>6)return false;
  let edits=0;
  for(const clause of clauses){
    if(policy.actions.some(p=>new RegExp('^(?:'+p+')$').test(clause))){edits++;continue;}
    if(![...policy.preservation,...policy.context].some(p=>new RegExp('^(?:'+p+')$').test(clause)))return false;
  }
  return edits>0&&edits<=3;
}

File: roundtrip/edit_policy.py
"""The same fail-closed Lightroom request grammar used by the hosted form."""
import json
import re
from pathlib import Path
POLICY=json.loads((Path(__file__).parent/'relay/lib/edit-policy.json').read_text())
def supported_edit(feedback):
    clauses=[s.strip() for s in re.split(r"[.!;]+|,?\s+but\s+|,\s*(?=keep\b)",re.sub(r'^please\s+','',feedback.lower().strip())) if s.strip()]
    if not clauses or len(clauses)>6:return False
    edits=0
    for clause in clauses:
        if any(re.fullmatch(p,clause) for p in POLICY['actions']):edits+=1
        elif not any(re.fullmatch(p,clause) for p in POLICY['preservation']):return False
    return 0<edits<=3
